src/dmd/lexer.d

```diff
@@ -202,6 +202,129 @@ unittest
     }
 }
 
+
+/**************
+ * Pool allocator for Token objects, used by the Lexer.
+ * Tokens are never freed so Token pointers will never change
+ */
+private struct TokenPool{
+private:
+    enum PoolSize = 1024; //number of tokens per pool
+    enum DefaultPoolListSize = 64; //number of pools to allocate by default
+    Token[][] pools;
+    size_t poolsAllocated = 0;
+
+    nothrow:
+
+    ///Token pool looks like a big array
+    public ref Token getByIndex(size_t index){
+        if (index >= PoolSize*poolsAllocated)
```

Comment:
Operators are preceded and succeeded by a single space. This is erratic throughout these changes.